server/web_ui/src/lib.rs

```diff
@@ -35,3 +39,52 @@ pub fn run_app() -> Result<(), JsValue> {
     gloo::console::debug!(kanidm_proto::utils::get_version("kanidmd_web_ui"));
     Ok(())
 }
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum RequestMethod {
+    GET,
+    POST,
+    PUT,
+}
+
+impl ToString for RequestMethod {
+    fn to_string(&self) -> String {
+        match self {
+            RequestMethod::PUT => "PUT".to_string(),
+            RequestMethod::POST => "POST".to_string(),
+            RequestMethod::GET => "GET".to_string(),
+        }
+    }
+}
+
+/// Build and send a request to the backend, with some standard headers and pull back
+/// (kopid, status, json, headers)
+pub async fn do_request(
+    uri: &str,
+    method: RequestMethod,
+    body: Option<JsValue>,
+) -> Result<(Option<String>, u16, JsValue, Headers), FetchError> {
+    let mut opts = RequestInit::new();
+    opts.method(&method.to_string());
+    opts.mode(RequestMode::SameOrigin);
```

Comment:
I think we want opts.credentials(SameOrigin here?)

Reply:
negatory, from my PR comment 😄 

> where you see me remove opts.credentials(RequestCredentials::SameOrigin); it's because [that's the default](https://developer.mozilla.org/en-US/docs/Web/API/Request/credentials)

Reply:
Ahhh. Even if it's the default, is there a harm in being explicit?

Reply:
Sure, no harm at all